### PR TITLE
set up RingMenu and InputListener

### PR DESCRIPTION
**Pull request by EstevanBR**

### CODE REVIEW

File: Input.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    inputStruct.downPressed = false;
+    inputStruct.leftPressed = false;
+    inputStruct.rightPressed = false;
+
+    inputStruct.upReleased = false;
+    inputStruct.downReleased = false;
+    inputStruct.leftReleased = false;
+    inputStruct.rightReleased = false;
+}
+void Input::logForInput() {
```
Comment (gillett-hernandez) [inline Input.cpp]: i feel like this function should have a more clearly defined role, or needs to be renamed. i.e. it is called logForInput yet it has a debug flag check and it calls other functions that are seemingly unrelated to logging

Comment (EstevanBR) [inline Input.cpp]: Which one? clear or log?

Comment (EstevanBR) [inline Input.cpp]: @gillett-hernandez check it now.

**Comment by gillett-hernandez** [comment]
bruh just merge it

**Pull request merged**